src/comparable.ml

```diff
@@ -0,0 +1,13 @@
+module type S = sig
+  type t
+
+  include Equatable.S with type t := t
+
+  val compare:t Comparator.t
+end
+
+module Default(T: sig type t end) : S = struct
```

Comment:
Provide a standard comparators and equality functions for all the built in types (int, string, etc). Authors should roll their own comparison and equality functions for their types. Polymorphic compare requires walking the object tree at runtime.  See: https://janestreet.github.io/ocaml-perf-notes.html

Ideally we could use the type information to generate comparison functions per type at compile time (maybe using a PPX transform) that uses the type info to generate a static comparator for record and union types. This is exactly how F# works.